WBooks/AppDelegate.swift

```diff
@@ -7,33 +7,74 @@
 //
 
 import UIKit
+import GoogleSignIn
 
 @UIApplicationMain
-class AppDelegate: UIResponder, UIApplicationDelegate {
-
+class AppDelegate: UIResponder, UIApplicationDelegate, GIDSignInDelegate {
```

Comment:
It's better to leave the App Delegate as "dumb" as possible, because it's the entry point to many different actions. So it's better that this has no other responsibility other than redirect to the one who knows how to react. So it shouldn't be delegate of Google Sign In.

So the delegate should be someone who knows what to do. At first sight, the view controller or view model of the view that triggers the login could be the delegate, because he knows that a login will be triggered and knows what to do if there is an error (for example, show an alert with the error) or if there is not (for example, move to next view). On another thought, this GoogleSignIn is a service, which is "independent". If we need to add logic to the framework's service, then we add our own wrapper to the service, and have the one that needs it handle it (the login view related components). If we don't need to add logic, then it's unnecessary.